streamz/core.py

```diff
@@ -1637,15 +1638,20 @@ class flatten(Stream):
     """
     def update(self, x, who=None, metadata=None):
         L = []
-        for i, item in enumerate(x):
-            if i == len(x) - 1:
-                y = self._emit(item, metadata=metadata)
-            else:
-                y = self._emit(item)
+        items = chain(x)
+        item = next(items)
+        for item_next in items:
+            y = self._emit(item)
+            item = item_next
             if type(y) is list:
                 L.extend(y)
             else:
                 L.append(y)
+        y = self._emit(item, metadata=metadata)
+        if type(y) is list:
```

Comment:
`isinstance(y, list)` ? Would any iterable work here?

Reply:
I took that from the previous implementation and wondered about the same question. As I wasn't sure if there is some special treatment for lists hidden somewhere else in the codebase, I decided to keep it as is.

Reply:
OK, well if it was like that before...